lms/djangoapps/courseware/tests/test_discussion_module.py

```diff
@@ -24,3 +26,57 @@ def test_html_with_user(self):
         html = fragment.content
         self.assertIn('data-user-create-comment="false"', html)
         self.assertIn('data-user-create-subcomment="false"', html)
+
+    def test_discussion_render_successfully_with_orphan_parent(self):
+        """
+        Test that discussion module render successfully
+        if discussion module is child of an orphan.
+        """
+        user = UserFactory.create()
+        store = modulestore()
+        course = store.create_course('testX', 'orphan', '123X', user.id)
+        orphan_sequential = store.create_item(self.user.id, course.id, 'sequential')
+
+        vertical = store.create_child(
+            user.id,
+            orphan_sequential.location,
+            'vertical',
+            block_id=course.location.block_id
+        )
+
+        discussion = store.create_child(
+            user.id,
+            vertical.location,
+            'discussion',
+            block_id=course.location.block_id
+        )
+
+        discussion = store.get_item(discussion.location)
+
+        grand_parent = self.get_grand_parent(discussion)
+        # Assert that orphan sequential is last grand parent of discussion module.
+        self.assertEqual(orphan_sequential.location, grand_parent.location)
+
+        discussion = get_module_for_descriptor_internal(
```

Comment:
@adampalay this ```discussion``` is with ```LmsModuleSystem``` and the above one is without we can't call render on above ```discussion```.